lib/dom.js

```diff
@@ -606,14 +606,27 @@ Document.prototype = {
 		return rtv;
 	},
 	
-	getElementsByClassName: function(className) {
-		var pattern = new RegExp("(^|\\s)" + className + "(\\s|$)");
+	getElementsByClassName: function(classNames) {
+		var spaceSeparatedTokens = function(string) {
+			// 5 space characters, as defined in html spec
+			return string.split(/[ \t\n\f\r]+/).filter(function(name) {
+				return name !== ''
+			})
+		}
```

Comment:
I think there is no need to redefine this method every time.
But since the "Infra Spec" [has an own section about this](https://infra.spec.whatwg.org/#split-on-ascii-whitespace), it's probably worth having this peace of code in an extra function on the top level of the module.

Please move it there, but use the following syntax and naming (function name and parameter name), instead of a `var` assignment:
```js
/**
 * @see https://infra.spec.whatwg.org/#split-on-ascii-whitespace
 */
function splitOnASCIIWhitespace (value) {
  // put the implementation here
}
```